packages/react-native-reanimated/src/animation/styleAnimation.ts

```diff
@@ -128,9 +128,11 @@ export function withStyleAnimation(
 
           // When working with animations changing colors, we need to make sure that each one of them begins with a rgba, not a processed number.
           // Thus, we only set the path to a processed color, but currentStyleAnimation.current stays as rgba.
-          const isAnimatingColorProp = ColorProperties.includes(
-            currentEntry.path[0] as string
-          );
+          const isAnimatingColorProp =
+            ColorProperties.includes(currentEntry.path[0] as string) ||
+            (currentEntry.path[0] === 'boxShadow' &&
+              currentEntry.path.length > 2 &&
+              currentEntry.path[2] === 'color');
```

Comment:
I think we should move this to a separate function because, in the future, this condition will likely grow, and we'll need to consider refactoring it into something more sophisticated than just a bunch of if statements to maintain readability.